main/util/HullPolygon.ts

```diff
@@ -19,7 +20,11 @@ export class HullPolygon {
   /**
    * Returns the signed area of the specified polygon.
    */
-  public static area(polygon: Polygon): number {
+  public static area(polygon: Polygon | PolygonArray): number {
```

Comment:
Would it make sense to represent various properties in the type system? For example, we could have HullPolygon be a subclass of Polygon, and AxisAlignedRectangle as a subclass of HullPolygon. This would make some of this code simpler or easier to follow, hull() can return HullPolygon and so on. We can still override methods as needed for precision or speed, since eg area of AxisAlignedRectangle is very simple.